Source/Core/DiscIO/WIABlob.h

```diff
@@ -34,7 +34,27 @@ enum class WIARVZCompressionType : u32
   Zstd = 5,
 };
 
-std::pair<int, int> GetAllowedCompressionLevels(WIARVZCompressionType compression_type);
+template <bool gui>
+std::pair<int, int> GetAllowedCompressionLevels(WIARVZCompressionType compression_type)
+{
+  switch (compression_type)
+  {
+  case WIARVZCompressionType::Bzip2:
+  case WIARVZCompressionType::LZMA:
+  case WIARVZCompressionType::LZMA2:
+    return {1, 9};
+  case WIARVZCompressionType::Zstd:
+    // The actual minimum level can be gotten by calling ZSTD_minCLevel(). However, returning that
+    // would make the UI rather weird, because it is a negative number with very large magnitude.
+    // Note: Level 0 is a special number which means "default level" (level 3 as of this writing).
+    if (gui)
+      return {1, ZSTD_maxCLevel()};
+    else
+      return {ZSTD_minCLevel(), ZSTD_maxCLevel()};
```

Comment:
Sorry for not catching this earlier, but this requires `zstd.h` to be included. It may be better to just keep this function in the .cpp file and add a regular parameter instead of making this function templated, so that we don't pollute too much with includes. This function is called very infrequently, so any minor performance difference there would be is not a concern.